src/crypto/kdf/AesKdf.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (C) 2017 angelsl
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <format/KeePass2.h>
+#include <QtConcurrent>
+#include <crypto/SymmetricCipher.h>
+#include <crypto/CryptoHash.h>
+#include <crypto/Random.h>
+#include "AesKdf.h"
+
+void AesKdf::randomizeSalt(QVariantMap &p) {
+    p.insert(KeePass2::KDFPARAM_AES_SEED, randomGen()->randomArray(32));
+}
+
+QByteArray AesKdf::transform(QByteArray key, QVariantMap p, bool* ok, QString* errorString) {
```

Comment:
This is very ugly. I know you only copied the existing code, but when you are already on it, please refactor.

You should use a non-const reference parameter for the resulting QByteArray and use ok as function return value. Pointer to error string is also bad style and can go terribly wrong. If there can be different error codes, use a status enum as return value instead of a bool.

Same for the Argon KDF.